src/java/org/apache/cassandra/index/sai/disk/v1/MemtableIndexWriter.java

```diff
@@ -55,19 +55,25 @@ public class MemtableIndexWriter implements PerColumnIndexWriter
     private static final Logger logger = LoggerFactory.getLogger(MemtableIndexWriter.class);
 
     private final IndexDescriptor indexDescriptor;
-    private final IndexContext indexContext;
+    private final IndexTermType indexTermType;
+    private final IndexIdentifier indexIdentifier;
+    private final IndexMetrics indexMetrics;
```

Comment:
nit: Another one of these cases where I guess we could just pass and store the SAI...if it doesn't make testing harder...

Reply:
My intent here had always been to only pass references to the classes that were being used. Passing a `StorageAttachedIndex` makes sense in the case of `IndexViewManager` because the index is needed to create the `SSTableIndex`. In this case we don't need an index.